src/components/Header/Header.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { FormEvent, useEffect, useRef } from 'react';
+
+type Props = {
+  onAddTodo: (title: string) => void,
+  todoTitle: string,
+  setTodoTitle: (todoTitle: string) => void,
+  isBeingLoading: boolean,
+  showExpendIcon: boolean,
+  onStatusAll: () => void,
+};
+
+export const Header: React.FC<Props> = ({
+  onAddTodo,
+  todoTitle,
+  setTodoTitle,
+  isBeingLoading,
+  showExpendIcon,
+  onStatusAll,
+}) => {
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    onAddTodo(todoTitle);
+  };
+
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [isBeingLoading]);
+
+  return (
+    <header className="todoapp__header">
+      {showExpendIcon
+        && (
+          // eslint-disable-next-line jsx-a11y/control-has-associated-label
+          <button
```

Comment:
You should add a label so that the linter doesn't hide the error

Reply:
strongly agree